crates/dbsp/src/storage/backend/posixio_impl.rs

```diff
@@ -73,15 +75,38 @@ impl FileReader for PosixReader {
         self.drop.keep();
     }
 
-    fn read_block(&self, offset: u64, size: usize) -> Result<Arc<FBuf>, StorageError> {
-        let mut buffer = FBuf::with_capacity(size);
+    fn read_block(&self, location: BlockLocation) -> Result<Arc<FBuf>, StorageError> {
+        let mut buffer = FBuf::with_capacity(location.size);
 
-        match buffer.read_exact_at(&self.file, offset, size) {
+        match buffer.read_exact_at(&self.file, location.offset, location.size) {
             Ok(()) => Ok(Arc::new(buffer)),
             Err(e) => Err(e.into()),
         }
     }
 
+    fn read_async(
+        &self,
+        blocks: Vec<std::ops::Range<u64>>,
+        callback: Box<dyn FnOnce(Vec<Result<Arc<FBuf>, StorageError>>) + Send>,
+    ) {
+        let file = self.file.clone();
+        TOKIO.spawn_blocking(move || {
```

Comment:
is this ok? does it spawn a thread kn every read?

Reply:
spawn_blocking uses a Tokio thread pool, so in steady state it shouldn't spawn a new thread.

Even if you use tokio's built-in async read/write, they just boil down to this same thing, because there's no other way to do async file i/o on Linux other than io_uring (or io_setup(2) and friends, which are *less* widely supported than io_uring.)

I'm only planning to use this in places where we might have several buffers.